analyses/independent-samples/02-generate-independent-rnaseq.R

```diff
@@ -83,7 +83,7 @@ independent_rna_primary_plus_polya <- sample_df %>%
                           histology_df = .,
```

Comment:
Hi @runjin326 . I have a dplyr question about this block of code (listed below). Sorry that I am not very familiar with dplyr.

<https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d1e4240c51949216e382a8f782d5314fb8f1b780/analyses/independent-samples/02-generate-independent-rnaseq.R#L77-L86>

Will the filtered `sample_df` be passed to `independent_rna_samples` as `histology_df`? If so, the `histology_df` only has `poly-A` samples. Then, how can `independent_dna` still be obtained by the following code? <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d1e4240c51949216e382a8f782d5314fb8f1b780/analyses/independent-samples/independent_rna_samples.R#L45-L48>

Reply:
Hi @logstar I actually wrote the above code for OpenPBTA so I can probably answer these questions. 

Just to clarify ,you wanted to understand
- how `independent_rna_samples` is able to read the filtered sample_df for polya samples? 
- and the second is how `independent_dna` gets the assigned the filtered histology_df values?

Reply:
Hello Yuanchao,
Yes the results of ```
sample_df %>% 
   filter(sample_type == "Tumor", RNA_library=="poly-A", 
          composition == "Solid Tissue" 
   ) 
```will be passed to the function `independent_rna_samples`. 
For finding independent_dna, it is using the Kids_First_Biospecimen_ID as a filter, basically asking whether any of these specimens also has DNA results (which will be captured in the independent specimen list). I hope that answers your question.

Reply:
@kgaonkar6 Thank you for the quick reply. Yes.

> * how `independent_rna_samples` is able to read the filtered sample_df for polya samples?

I assume this is done by the `.` in `histology_df = .`.

> * and the second is how `independent_dna` gets the assigned the filtered histology_df values?

In the `independent_rna_samples` function call, if the `histology_df` is the filtered `sample_df` with only `poly-A` samples, no `histology_df$Kids_First_Biospecimen_ID` should be in `independent_dna_sample_df$Kids_First_Biospecimen_ID`, assuming that `RNA_library=="poly-A"` and `experimental_strategy=="WGS"` are mutually exclusive.

Reply:
> Hello Yuanchao,
> Yes the results of `sample_df %>% filter(sample_type == "Tumor", RNA_library=="poly-A", composition == "Solid Tissue" ) `will be passed to the function `independent_rna_samples`.
> For finding independent_dna, it is using the Kids_First_Biospecimen_ID as a filter, basically asking whether any of these specimens also has DNA results (which will be captured in the independent specimen list). I hope that answers your question.

Hi @runjin326 . Thank you for the explanation. It answers my first question about passing the left hand side value using `.`.

Regarding the `independent_dna`, I was wondering whether the `independent_dna` will be an empty `tibble`. Sorry for the confusion in my first comment. As the `histology_df` only has `poly-A` samples, I assume no sample should have `Kids_First_Biospecimen_ID` in the `independent-specimens.wgswxspanel.primary-plus.tsv`. The `independent-specimens.wgswxspanel.primary-plus.tsv` is passed to the R script by the following code:

<https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/75cae1274e35ad2a585743338206d6df52ffc9a4/analyses/independent-samples/run-independent-samples.sh#L30-L34>

Reply:
@logstar you are right! Thanks for catching that, this

<https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d1e4240c51949216e382a8f782d5314fb8f1b780/analyses/independent-samples/independent_rna_samples.R#L45-L48>

should be the following to match the Participant IDs instead of the Biospecimen IDs
```
 independent_dna <- histology_df %>%
    # include matched independent_dna samples
    dplyr::filter(Kids_First_Participant_ID %in%
                    independent_dna_sample_df$Kids_First_Participant_ID)
```

Reply:
> @logstar you are right! Thanks for catching that, this
> 
> https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d1e4240c51949216e382a8f782d5314fb8f1b780/analyses/independent-samples/independent_rna_samples.R#L45-L48
> 
> should be the following to match the Participant IDs instead of the Biospecimen IDs
> 
> ```
>  independent_dna <- histology_df %>%
>     # include matched independent_dna samples
>     dplyr::filter(Kids_First_Participant_ID %in%
>                     independent_dna_sample_df$Kids_First_Participant_ID)
> ```

Thank you for the modification! It is very clear to me now.

Reply:
I will modify the code accordingly ;) Thank you both!